common/script/libs/shops.js

```diff
@@ -0,0 +1,219 @@
+import _ from 'lodash';
+import content from '../content/index';
+import i18n from '../../../common/script/i18n';
+
+let shops = {};
+
+function lockQuest (quest, user) {
+  if (quest.lvl && user.stats.lvl < quest.lvl) return true;
+  if (user.achievements.quests) return quest.previous && !user.achievements.quests[quest.previous];
+  return quest.previous;
+}
+
+shops.getMarketCategories = function getMarket (user, language) {
+  let categories = [];
+  let eggsCategory = {
+    identifier: 'eggs',
+    text: i18n.t('eggs', language),
+    notes: i18n.t('dropsExplanation', language),
+  };
+  eggsCategory.items = _(content.questEggs).values().filter(egg => egg.canBuy(user)).concat(_.values(content.dropEggs)).map(egg => {
```

Comment:
I think splitting this into multiple lines will be enough, without creating new variables